components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
 
-			try {
-
-				WhiteListDTO whiteListDTO = new WhiteListDTO();
-				whiteListDTO.setApiID(apiID);
-				whiteListDTO.setApplicationID(applicationID);
-				//whiteListDTO.setSubscriptionID(subscriptionID);
-				whiteListDTO.setUserMSISDN(msisdn);
+		APIBlacklistWhitelistResponseDTO response;
+		BlacklistWhitelistDTO blacklistWhitelistDTO = new BlacklistWhitelistDTO();
+		Gson gson = new Gson();
+		List<String> msisdnList = new ArrayList<>();
+		BufferedReader br;
+		int limit = 5000; //max number of records to process
 
-				blackListWhiteListService.whiteListSubscription(whiteListDTO);
+		app = replaceAllWithWildcard(app);
+		api = replaceAllWithWildcard(api);
+		sp = replaceAllWithWildcard(sp);
 
-				final StringBuilder succMSG = new StringBuilder();
+		blacklistWhitelistDTO.setAppId(app);
+		blacklistWhitelistDTO.setApiID(api);
+		blacklistWhitelistDTO.setAction(action);
+		blacklistWhitelistDTO.setServiceProvider(sp);
+		blacklistWhitelistDTO.setUser(user);
 
-				succMSG.append("{").append("\"Success\":").append("{").append("\"messageId\":\"")
-						.append("WhiteListed Number".intern()).append("\",");
-				succMSG.append("\"text\":\"").append("WhiteListed Number Successfully Added to the system ".intern())
-						.append("\",");
-				succMSG.append("\"variables\":\"").append(msisdn).append("\"").append("}}");
 
-				return Response.status(Response.Status.OK).entity(succMSG.toString()).build();
+		br = new BufferedReader(new InputStreamReader(uploadedInputStream, StandardCharsets.UTF_8));
 
-			}catch( InvalidMSISDNException e1){
-				return Response.status(Response.Status.BAD_REQUEST).entity(e1.getErrorType()).build();
+		try {
+			while (br.ready() && limit > 0) {
+				msisdnList.add(br.readLine());
+				limit--;
+			}
+			Response.Status status = Response.Status.CREATED;
+			blacklistWhitelistDTO.setMsisdnList(msisdnList);
+			response = blackListWhiteListService.blacklist(blacklistWhitelistDTO);
 
-			} catch (BusinessException e) {
-				return Response.status(Response.Status.BAD_REQUEST).entity(e.getErrorType()).build();
-			} catch (Exception e) {
-				e.printStackTrace();
+			if(response.getFailed() > 0 && response.getProcessed() == 0){
+				status = Response.Status.BAD_REQUEST;
 			}
-		}
-		else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(errorMSG.toString()).build();
+
+			return Response.status(status).entity(gson.toJson(response)).build();
+		} catch (IOException | BusinessException e) {
+			LOG.error("Unable to process upload", e);
+			return Response.status(500).build();
 		}
 
-		return Response.ok().build();
 	}
 
-	@POST
-	@Path("/Whitelist")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response bulkWhitelist(String jsonBody) throws SQLException {
-		LOG.debug("bulkWhitelist Triggerd  jsonBody :" + jsonBody);
-
+	/**
+	 * Removes from the blacklist
+	 *
+	 * { "apiName":"USSD" }
+	 */
+	@DELETE
+	@Path("/remove/{app}/{api}")
```

Comment:
avoid the "remove" text in the endpoint URL. HTTP request DELETE denotes that you are trying to remove.